### PR TITLE
fetch data from aptos sdk directly, instead of depending on external indexer

### PR DESCRIPTION
**Pull request by SamuelQZQ**

### CODE REVIEW

File: test/test-pools.json [hunk 1]
```diff
@@ -55,196 +55,224 @@
   ],
   "pools": [
     {
-      "name": "S:100:APT:tAPT",
```
Comment (0xbe1) [inline test/test-pools.json:58]: do we not have coin type in pool object?

**Review by 0xbe1 — APPROVED** [review]
lgtm

File: test/test-pools.json (hunk 1 above)
Comment (SamuelQZQ) [reply test/test-pools.json:58]: yes we have. The `poolType` here is parsed from pool object

File: src/PoolDataClient.ts [hunk 2]
```diff
@@ -1,39 +1,116 @@
...
+              asset_0: { value: string };
+              asset_1: { value: string };
+              asset_2: { value: string };
+              asset_3: { value: string };
+              amp_factor?: string;
             };
-          });
+          }[];
+
+          const allCoinAddress = uniq(
```
[the remark this answers is not in the document]
Comment (SamuelQZQ) [reply src/PoolDataClient.ts:52]: seems not much difference?

**Pull request merged**